packaging/build/Microsoft.NET.Build.Containers.targets

```diff
@@ -34,7 +34,7 @@
             <ContainerBaseImage Condition="'$(ContainerBaseImage)' == ''">$(_ContainerBaseRegistry)/$(_ContainerBaseImageName):$(_ContainerBaseImageTag)</ContainerBaseImage>
             <ContainerRegistry Condition="'$(ContainerRegistry)' == ''">docker://</ContainerRegistry>
             <!-- Note: spaces will be replaced with '-' in ContainerImageName and ContainerImageTag -->
-            <ContainerImageName Condition="'$(ContainerImageName)' == ''">$(AssemblyName)</ContainerImageName>
+            <ContainerImageName Condition="'$(ContainerImageName)' == ''">$(AssemblyName.ToLowerInvariant())</ContainerImageName>
```

Comment:
this shouldn't need to be done in the targets, we do the normalization in [this method](https://github.com/dotnet/sdk-container-builds/blob/tim-the-tooltask-taylor/Microsoft.NET.Build.Containers/ContainerHelpers.cs#L148-L166) - was there a reason that wasn't working?

Reply:
So the problem (from my perspective) was that running `msbuild /t:Publish /p:RuntimeIdentifier=linux-x64 /p:Configuration=Release /p:PublishProfile=DefaultContainer` is essentially saying "do all the defaults", but the default led to a capitalized assemblyname, which led to a warning in the "default" case.

Thoughts on this? Is the warning okay here?

Reply:
We decided to remove this from the targets, let the task normalize it and log a high-importance message instead of warning.

Reply:
Justification for this is that the task/targets should be _very_ focused on doing the right thing by default. Telling a user to do a thing when we could have just done it feels bad, especially when the response to us saying 'hey you can't have these characters' is going to be writing the same regex and replacements that we already do.

Reply:
I would ideally like to have a warning _if the user specified a bad image name_, but warningless correction _if we inferred the name ourselves_. Think that's worth going to pains for?

Reply:
This would require some type of flag set when we did inference that we'd pass into the tasks? so during the inference phase we'd set `<_ContainerDidInferImageName>true</_ContainerDidInferImageName>`, flow that through as a boolean input to both tasks, and then pass it into the validation phase to let us know whether to warn or not?